### DISCUSSION TITLE
Vertx Gateway Verticles Configurations

### DISCUSSION THREAD

**Discussion by shekarnode** [apiman/apiman#1350] [Help]
Hi Marc,

I have increased the verticle to 24 in config file , but i could still see only 8 parallel thread has been created.
So are these verticle = vertx event loops thread ?

![image](https://user-images.githubusercontent.com/3267543/123519324-1b9cdb80-d6c8-11eb-9d32-f807927be37e.png)

**Comment by msavy** [comment]
Did you see logging statement indicating that additional verticles were being created?

**Reply by msavy** [reply]
For example, I set my HTTP verticle count to be `10`, and I see:

```
❯ ./apiman-gateway.sh --conf configs/conf-es.json
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2021-07-04 08:39:59.767 [vert.x-eventloop-thread-0] InitVerticle - Starting verticle: INITIALISER. UUID: 95564be8-eff5-44c0-9b37-ee4ee2df817c.INITIALISER.
[INFO ] 2021-07-04 08:39:59.771 [vert.x-eventloop-thread-0] InitVerticle - Will deploy 1 of type API
[INFO ] 2021-07-04 08:39:59.775 [vert.x-eventloop-thread-0] InitVerticle - Will deploy 10 of type HTTP
[INFO ] 2021-07-04 08:39:59.775 [vert.x-eventloop-thread-1] ApiVerticle - Starting verticle: API. UUID: c454e903-06b4-48c9-8b46-c9227e15c225.API.
[INFO ] 2021-07-04 08:39:59.778 [vert.x-eventloop-thread-0] InitVerticle - Will deploy 0 of type HTTPS
[INFO ] 2021-07-04 08:39:59.778 [vert.x-eventloop-thread-2] HttpGatewayVerticle - Starting verticle: HTTP. UUID: b3eff37e-6817-47d3-b6bc-a852547de5ec.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-5] HttpGatewayVerticle - Starting verticle: HTTP. UUID: e91a90cc-bf15-4865-9b07-6fee8750cc54.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-6] HttpGatewayVerticle - Starting verticle: HTTP. UUID: 686a4981-b1fc-404f-b5e5-3605247374f3.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-3] HttpGatewayVerticle - Starting verticle: HTTP. UUID: 3462b1a6-cb8b-4e15-9a3f-b59780a08f2c.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-4] HttpGatewayVerticle - Starting verticle: HTTP. UUID: dc507e98-8d5a-478a-bb83-12c208eb49e3.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-10] HttpGatewayVerticle - Starting verticle: HTTP. UUID: ec4aacf3-18ea-458d-a697-43623a488dec.HTTP.
[INFO ] 2021-07-04 08:39:59.779 [vert.x-eventloop-thread-9] HttpGatewayVerticle - Starting verticle: HTTP. UUID: dbd81960-cf16-4048-93b9-c0c7b6d74b87.HTTP.
[INFO ] 2021-07-04 08:39:59.780 [vert.x-eventloop-thread-7] HttpGatewayVerticle - Starting verticle: HTTP. UUID: 8fb2ddeb-0807-4ea9-8605-1b0ecc2f1af8.HTTP.
[INFO ] 2021-07-04 08:39:59.780 [vert.x-eventloop-thread-8] HttpGatewayVerticle - Starting verticle: HTTP. UUID: 1aef7522-56d2-4fdf-9b79-4756a4f4ccb4.HTTP.
[INFO ] 2021-07-04 08:39:59.782 [vert.x-eventloop-thread-11] HttpGatewayVerticle - Starting verticle: HTTP. UUID: abecabc0-dc6b-48fa-a256-c004c1a78536.HTTP.
[INFO ] 2021-07-04 08:39:59.875 [vert.x-eventloop-thread-1] Reflections - Reflections took 72 ms to scan 1 urls, producing 10 keys and 23 values
```

**Reply by shekarnode** [reply]
Hi marc,

I have tried to achieve the same but now it is showing 24 verticals deployed message.
but i could not see that many verticles

![image (4)](https://user-images.githubusercontent.com/3267543/126606811-09c7d167-bc5f-4a66-813c-df481ed9b0df.png)


So my question is verticles = thread ?

**Reply by msavy** [reply]
In general, no, each verticle does not create a new thread. Please read: https://vertx.io/docs/vertx-core/java/#_reactor_and_multi_reactor

The default from Vert.x is `DEFAULT_EVENT_LOOP_POOL_SIZE = 2 * CpuCoreSensor.availableProcessors()`. That may include SMT (hyperthreading).

Let's take a step back for a moment. What are you trying to achieve that you think increasing the number of threads in the event loop pool will help? Why do you want more event loop threads?

**Reply by shekarnode** [reply]
Actually I thought increasing threads would improve performance. As you quoted about default verticle , I have 4 core CPU so 8 verticles make sense. Please excuse me for this question  , I did not have much prior knowledge about this. I would go through link which you have shared for my better understanding. Thanks for the help.

**Comment by msavy** [comment]
Are you having some performance issues? Are you getting warnings about threads being blocked?
